### ISSUE TITLE
Installation error (min gcc version required ) 

### ISSUE THREAD

**Issue by IsmailM** [heathsc/gemBS#74]
Hey,

Am currently trying to install the latest version of GemBS, but getting the following error:

```
make[1]: Entering directory `/mnt/254b78b9-76b4-422d-84b1-cc632bff60f7/IsmailM/.progz/gemBS/tools/utils'
gcc -O3 -g -I. -Icommon     -o ../bin/gemBS_cat gemBS_cat.c common/utils.c   -lm
gemBS_cat.c: In function ‘main’:
gemBS_cat.c:22:2: error: ‘for’ loop initial declarations are only allowed in C99 mode
  for(int ix = 1; ix <= argc; ix++) {
  ^
gemBS_cat.c:22:2: note: use option -std=c99 or -std=gnu99 to compile your code
make[1]: *** [../bin/gemBS_cat] Error 1
make[1]: Leaving directory `/mnt/254b78b9-76b4-422d-84b1-cc632bff60f7/IsmailM/.progz/gemBS/tools/utils'
make: *** [_utils] Error 2
```

I have `gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)` installed

Is there an easy fix - What is the version of gcc required?

Setting `CFLAGS` as follows did not fix the issue: `CFLAGS=-std=gnu99 python setup.py install --user`

**Comment by AndyWangSFU** [comment]
I met the same error before. There is a very direct way but I am not sure whether it is "easy fix" for you or not. I manually changed every:

`for (int ix = 1; ....)`
 to 
```
int ix; 
for (ix = 1; ....)
```
What I know from this bug is just C interpreter / compiler issues: you cannot declare a variable inside a loop, which I feel it stupid. 

Note: it may take you a while because this issue occurs in a few .c files. For instance, you can see that this error currently occurs in `gemBS_cat.c` and `utils.c`. What you need to do is just open each file, search the keyword `for (int` and manually change it. Then rerun the installation step (I remember it occurs in `make` step?), until every .c file fixed.

Hopefully it helps!

**Comment by IsmailM** [comment]
Many Thanks @AndyWangSFU 

Being lazy, I found a slightly easier way 🙃 

I added `-std=gnu99` to the end of the two following lines

https://github.com/heathsc/gemBS/blob/522d41a6624b97ea406962fb236523ad4ba77275/tools/utils/Makefile.in#L20

https://github.com/heathsc/gemBS/blob/522d41a6624b97ea406962fb236523ad4ba77275/tools/utils/Makefile.in#L23

Would be great if $CFLAGS was used here. 👍

**Comment by heathsc** [comment]
Good point - I should check if an old compiler is being used and set the
flags accordingly.

Simon

**Comment by heathsc** [comment]
FIxed in 3.5.1.  The configure script now checks if --std=c99 is required
for standard conformant behaviour.
